integration/test/ParseLiveQueryTest.js

```diff
@@ -32,6 +33,35 @@ describe('Parse LiveQuery', () => {
     object.set({ foo: 'bar' });
     await object.save();
   });
+  
+  it('can subscribe to matchesKeyInQuery', async done => {
+    const objectA = new TestObject();
+    const objectB = new DiffObject();
+    const objectC = new SampleObject();
+
+    objectA.set({ name: 'test', sample: objectC });
+    objectB.set({ name: 'diff', sample: objectC });
+    objectC.set({ name: 'sample' });
+
+    await Parse.Object.saveAll([objectA, objectB, objectC]);
+
+    const subQuery = new Parse.Query(DiffObject);
+    subQuery.equalTo('name', 'diff');
+
+    const query = new Parse.Query(TestObject);
+    query.contains('name', 'test');
+    query.matchesKeyInQuery('sample', 'sample', subQuery);
+
+    const subscripton = await query.subscribe();
+
+    subscripton.on('update', object => {
+      assert.equal(object.get('name'), 'tester');
+      done();
+    });
+
+    objectA.set({ name: 'tester' });
+    await objectA.save();
+  });
```

Comment:
Saving object A with tester doesn’t trigger the live query. 

The trigger occurs when the save matches the query you subscribe to. 

I can look more in-depth shortly

Reply:
I changed the event to `leave`